src/main/webapp/3Dscript.js

```diff
@@ -0,0 +1,201 @@
+}
+
+
+/**
+ * This function fetchs pose data from the RunInfo servlet,
+ * it is an asynchronous call requiring addMap() to be 
+ * called after the data is fully loaded.
+ */
+function fetchData() {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const id = urlParams.get('id');
+    const type = urlParams.get('dataType');
+    fetch('/getrun?id=' + id + '&dataType=' + type)
+    .then(response => response.json())
+    .then(data => pose = data)
+    .then(() => addMap());
+}
+
+/**
+ * Creates the 2D terrain and points a light at it.
+ * It also calls two other functions so it can wait on the fetch 
+ * request and significantly cut down rendering time with 
+ * geometry instancing.
+*/
+function addMap() {
+  // Creates the 2D map.
+  let loader = new THREE.TextureLoader();
+  let material = new THREE.MeshLambertMaterial({ 
+    map: loader.load(
+      'https://maps.googleapis.com/maps/api/staticmap' +
+      '?format=png&center=' + pose[0].lat + ',' + pose[0].lng + 
+      '&zoom=18&size=500x500&key=' + apiKey)
+    });
+  let geometry = new THREE.PlaneGeometry(50,50);
+  const map = new THREE.Mesh(geometry, material);
+  map.position.set(0,-4,0);
+  map.rotation.x = -Math.PI / 2;
+  scene.add(map);
+
+  // Add the light to the scene.
+  let light = new THREE.PointLight(0xffffff, 1, 0 );
+  light.position.set(0, 100, 0);
+  scene.add(light)
+  plotTrajectory();
+  
+  // Adds pose data to the scene as one Geometry instance.
+  material = new THREE.MeshBasicMaterial({color: 'red'});
+  geometry = new THREE.CylinderBufferGeometry(.005, .005, .1);
+  orientation = new THREE.InstancedMesh(geometry, material, pose.length);
+  orientation.instanceMatrix.setUsage(THREE.DynamicDrawUsage);
+  scene.add(orientation);
+  plotOrientation();
+}
+
+/**
+ * This uses the pose data to create a blue line representing the pose
+ * trajectory. 
+ */
+function plotTrajectory() {
+  // Removes any existing trajectory objects for repositioning.
+  scene.remove(trajectory);
+  /**
+   * The x axis controls the left and right direction, the y axis controls
+   * up and down movement, and the z axis controls forward and back movement.
+   */
+  let coordinates=[];
+  for (point of pose) {
+    coordinates.push(new THREE.Vector3(
+      /**
+       * Subtracting each point by the first point starts the pose
+       * at coordinates (0, 0, 0). Each unit in the 3D scene is 4
+       * meters while the 6th decimal point in GPS coordinates represents
+       * .11 meters. This is why we multiple the lat/lng by 25000, an
+       * increment in the 6th decimal place equates to a 0.025 unit
+       * change in our 3D space. Since altitude is already represented in
+       * meters, we simply divide by 4 to adjust for our 1:4 unit ratio.
+       */
+      (point.lng - pose[0].lng) * 25000 * poseTransform.scale,
+      (point.alt - pose[0].alt)/4,
+      (point.lat - pose[0].lat) * 25000 * -poseTransform.scale)
```

Comment:
So it is approximately convert from degrees to meters in local pose frame which the origin is the first pose? 
I think you can encapsulate it into a function e.g. LLADegreeToLocal(lat, lng, alt, origin), when we support to convert to ECEF, you can easily switch it to LLADegreeToECEF.